### PR TITLE
Reset to upstream v4.2.8 with select changes

### PR DESCRIPTION
**Pull request by toufali**
The base branch `upstream-v4.2.8` is identical to the upstream branch at `v4.2.8`.  

The branch being pulled, `upstream-v4.2.8-main-changes` is based off `main`.  It should include all the changes we intend to keep with our fork.  It should be reviewed and unnecessary changes should be identified and removed.

After merging this PR, the base branch (`upstream-v4.2.8`) should be merged to `main`.  Our main branch should then be as close to upstream as possible with our required additions.

### CODE REVIEW

**Comment by toufali** [comment]
I've done a first pass at reverting the more obvious stuff (code format, etc).

File: app/models/status.rb
Comment (toufali) [inline app/models/status.rb:1]: Let's investigate this code addition.

File: config/environments/production.rb
Comment (toufali) [inline config/environments/production.rb:1]: investigate this change

File: config/initializers/doorkeeper.rb [hunk 1]
```diff
@@ -174,8 +176,10 @@
   # Under some circumstances you might want to have applications auto-approved,
   # so that the user skips the authorization step.
   # For example if dealing with a trusted application.
+  redirect_uri_whitelist = ENV['REDIRECT_URI_WHITELIST']&.split('|').freeze
```
Comment (toufali) [inline config/initializers/doorkeeper.rb:179]: This allows us to use Elk

File: app/models/status.rb [hunk 2]
```diff
@@ -256,7 +256,7 @@ def non_sensitive_with_media?
   end
 
   def reported?
-    @reported ||= Report.where(target_account: account).unresolved.where('? = ANY(status_ids)', id).exists?
+    @reported ||= Report.where(target_account: account).where('? = ANY(status_ids)', id).exists?
```
Comment (toufali) [inline app/models/status.rb:259]: Let's investigate this code addition.

File: config/environments/production.rb [hunk 3]
```diff
@@ -72,7 +72,7 @@
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
   # English when a translation cannot be found).
-  config.i18n.fallbacks = [:en]
+  config.i18n.fallbacks = true
```
Comment (toufali) [inline config/environments/production.rb:75]: investigate this change – likely not required

File: app/models/status.rb (hunk 2 above)
Comment (aaga) [reply app/models/status.rb:259]: actually it was a code removal that I did (there have been no changes to upstream on this line).

It's needed in order to access private statuses and direct messages from Cinder.

I made a comment about it here in my PR on upstream: [link](https://github.com/mastodon/mastodon/pull/27675/files#r1382155905)

We could add some comment about it directly in the code here if that would be helpful for future

**Review by aaga — APPROVED** [review]
lgtm! love this format of merging in upstream.

**Pull request merged**